examples/rpc-custom-api/src/example.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * EXAMPLE
+ * Add a custom JSON RPC method to the base Solana JSON RPC API using @solana/web3.js.
+ *
+ * To run this example, execute `pnpm start` in this directory.
+ */
+import createLogger from '@solana/example-utils/createLogger.js';
+import {
+    Address,
+    address,
+    createDefaultRpcTransport,
+    createRpc,
+    createSolanaRpcApi,
+    DEFAULT_RPC_CONFIG,
+    mainnet,
+    RpcApi,
+    RpcApiMethods,
+    SolanaRpcApiMainnet,
+} from '@solana/web3.js';
+
+const log = createLogger('Custom JSON RPC API');
+
+/**
+ * SETUP: CUSTOM JSON RPC API CALL SIGNATURE
+ * Define the call signature of the custom API. For this example we will use the Triton One
+ * `getAsset` API, available on the public mainnet RPC server.
+ * https://docs.triton.one/digital-assets-api/metaplex-digital-assets-api/get-asset
+ */
+type AssetMetadata = Readonly<{
+    description: string;
+    name: string;
+    symbol: string;
+}>;
+interface TritonGetAssetApi extends RpcApiMethods {
+    /**
+     * Define the ideal developer-facing API as a TypeScript type. Doing so will enable typechecking
+     * and autocompletion for it on the RPC instance.
+     */
+    getAssetMetadata(address: Address): AssetMetadata;
+}
+
+/**
+ * SETUP: CUSTOM JSON RPC API IMPLEMENTATION
+ * Create an instance of the default JSON RPC API, then create a wrapper around it to intercept
+ * calls for custom API methods. The wrapper should format the inputs to satisfy the API of the JSON
+ * RPC server, and post-process the server response to satisfy the call signature defined above.
+ */
+const solanaRpcApi = createSolanaRpcApi<SolanaRpcApiMainnet>(DEFAULT_RPC_CONFIG);
+/**
+ * Create a proxy that wraps the Solana RPC API and adds extra functionality.
+ */
+const customizedRpcApi = new Proxy(solanaRpcApi, {
```

Comment:
Don't we have a `createRpcApi` helper that creates a proxy for you?

Reply:
We do, but it doesn't let you fiddle with method name the way this example is 

It does let you do per-method `paramsTransformer` and `responseTransformer` though, so I wonder if we could just extend it to also do a `methodNameTransformer`? I think that's the only thing this example does that we can't do with the existing `createRpcApi`.

Reply:
Yeah, the point here was to show how you can layer _new_ methods onto the default RPC, as you would if you were using a server that supports a superset of the Solana RPC API (Triton, Quicknode, Helius, basically every one).

Reply:
The reason this is preferable to creating a new RPC instance is that they target the same server, so you want to share the connection `undici.Dispatcher`.

Reply:
Just found another place where we claim this to be exported.

https://github.com/solana-labs/solana-web3.js/blob/master/packages/library/README.md?plain=1#L391

Reply:
RE the proxy vs the `createRpcApi` helper I think that's fine. IIRC we have other examples in READMEs that use the `createRpcApi` helper to add simpler methods.